### DISCUSSION TITLE
Error executing command as another user: Not authorized

### DISCUSSION THREAD

**Discussion by andy5995** [mesonbuild/meson#12874] [Q&A] [answered]
When I try to install in a GitHub workflow

Omg, why?

```
Error executing command as another user: Not authorized

This incident has been reported.
```

I tried `meson install --skip-subprojects` and `ninja install`. I'm reasonably certain I'm not trying to install to a directory I don't have access to, and even if I was, it shouldn't say I'm attempting to run a command as another user.

Full workflow: https://github.com/netpanzer/netpanzer/blob/01e61d0807e86b33be96344340d7675a81ef6421/.github/workflows/c-cpp.yml

[logs_2310.zip](https://github.com/mesonbuild/meson/files/14316341/logs_2310.zip)

**Comment by dcbaker** [comment]
Because for reasons I don't understand if your run `ninja install` as a non-privlidiged user and attempt to install to a root location, then we try to pop up the window to type in your password for sudo.

**Reply by eli-schwartz** [reply]
I fixed that a long time ago but I think the version of meson in ancient Ubuntu is even older...

**Reply by eli-schwartz** [reply]
Also sudo doesn't have a window, we tried to pop up the window for *polkit*. Except now we use sudo and also check for an interactive console and also ask you if you want to do it and also print a lot of messages describing what in fact we are doing.

**Comment by eli-schwartz** [comment, marked as answer]
It looks like you're using GitHub Cache for the entire build directory. So running `meson setup` and passing the "prefix" option is having no effect.

What prefix was the cache originally configured with? I'm betting this is a bug you fixed when updating your CI configuration but the cache is preserving the bug.

Aside: this is why I would not cache the build directory, but instead use `ccache` and persist your CCACHE_DIR.

**Reply by andy5995** [reply]
I'm sure that must be it. Unless I've made some horrendous error with my variables or my knowledge, I'm not trying to install to any root location. I have a pending PR where I remove the cache so I think I'll just merge that, rebase, and try again. Thanks a heap!
